pylint/message/message_id_store.py

```diff
@@ -101,18 +102,25 @@ def _raise_duplicate_msgid(symbol: str, msgid: str, other_msgid: str) -> NoRetur
         )
         raise InvalidMessageError(error_message)
 
+    @functools.lru_cache()
     def get_active_msgids(self, msgid_or_symbol: str) -> List[str]:
-        """Return msgids but the input can be a symbol."""
-        # Only msgid can have a digit as second letter
-        is_msgid: bool = msgid_or_symbol[1:].isdigit()
-        msgid = None
-        if is_msgid:
+        """Return msgids but the input can be a symbol.
+
+        The cache has no limit as it will never go very high. It's a set number of
+        msgid/symbol pair from our own code and from user's checkers. How many msgid can
+        the user generate ? I guess it will never be more than 1000, and if each
+        msgid/symbol pair has around 50 characters, it's still ~= 5 kb to cache.
```

Comment:
```suggestion
        The cache has no limit as its size will likely stay minimal. For each message we store
        about 1000 characters, so even if we would have 1000 messages the cache would only
        take up ~= 1 Mb.
```

Reply:
Thanks for the clean synthesis of that word salad 😄

Reply:
Happy to help! 😄